syllabus/task_space/task_space.py

```diff
@@ -23,25 +31,60 @@ def __init__(self, gym_space: Union[Space, int], tasks=None):
 
     def _make_task_encoder(self, space, tasks):
         if isinstance(space, Discrete):
-            assert space.n == len(tasks), f"Number of tasks ({space.n}) must match number of discrete options ({len(tasks)})"
+            assert space.n == len(
+                tasks
+            ), f"Number of tasks ({space.n}) must match number of discrete options ({len(tasks)})"
             self._encode_map = {task: i for i, task in enumerate(tasks)}
             self._decode_map = {i: task for i, task in enumerate(tasks)}
-            encoder = lambda task: self._encode_map[task] if task in self._encode_map else None
-            decoder = lambda task: self._decode_map[task] if task in self._decode_map else None
+
+            def encoder(task):
+                return self._encode_map[task] if task in self._encode_map else None
+
+            def decoder(task):
+                return self._decode_map[task] if task in self._decode_map else None
+
         elif isinstance(space, Box):
-            encoder = lambda task: task if space.contains(np.asarray(task, dtype=space.dtype)) else None
-            decoder = lambda task: task if space.contains(np.asarray(task, dtype=space.dtype)) else None
+
+            def encoder(task):
+                return (
+                    task
+                    if space.contains(np.asarray(task, dtype=space.dtype))
+                    else None
+                )
+
+            def decoder(task):
+                return (
+                    task
+                    if space.contains(np.asarray(task, dtype=space.dtype))
+                    else None
+                )
+
```

Comment:
Did you run a linter on this or was the code really bothering you? Using functions instead of lambdas is probably an improvement

Reply:
Yes, my linter runs automatically on save so it might bring some unintentional modifications when I'm going through the project. Nothing personal against the lambdas 😁